client/src/modules/stock/entry/entry.js

```diff
@@ -31,6 +31,8 @@ function StockEntryController(
   const mapEntry = initEntryMap();
   const { params } = $state;
 
+  vm.fromShipmentRegistry = !!params.shipment;
```

Comment:
Here's a more elegant approach using ui-router:

First, inject the `$transition$` service.

Then:
```js
const isFromShipmentRegistry = $transition$.from().name === 'shipments';
```

This way, we don't have to worry about `params` and what they contain.

I would also encourage you to do this in the `submit` block, not here, since that is all you need it for.  You also don't need to attach it to the `vm` scope.

Reply:
Done.  Although I still needed to check whether a shipment UUID was provided since if we are in the shipments registry and we click on the Stock Exit page, $transition$ thinks we came from 'shipments'.  So I made the check (in Submit now) check both.   That approach seems more robust to future changes (eg, suppose in the future we jump to Stock Exit page with a shipment UUID, but not from the shipment registry, then we would not want to jump back to the shipment registry).

Reply:
That makes sense to me.